Index: src/main/java/Network/Client/ClientToServerCommunication.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/Network/Client/ClientToServerCommunication.java b/src/main/java/Network/Client/ClientToServerCommunication.java
new file mode 100644
--- /dev/null	(date 1715543528218)
+++ b/src/main/java/Network/Client/ClientToServerCommunication.java	(date 1715543528218)
@@ -0,0 +1,16 @@
+package Network.Client;
+
+import Network.Client.RMI.VirtualView;
+import exception.ChangedStateException;
+import exception.NotValidMoveException;
+import socket.Messages.Message;
+
+import java.rmi.RemoteException;
+
+public interface ClientToServerCommunication {
+    void connectToServer() throws Exception;
+    void update(Message message) throws Exception;
+    void connectClient(VirtualView client) throws RemoteException;
+
+    void setPlayerNumber(int playerNumber) throws RemoteException, NotValidMoveException, ChangedStateException;
+}
